modules/client-handler/src/main/java/org/apache/ignite/client/handler/JdbcQueryEventHandlerImpl.java

```diff
@@ -228,16 +227,25 @@ public CompletableFuture<JdbcBatchExecuteResult> batchPrepStatementAsync(
         });
     }
 
-    private CompletableFuture<Long> executeAndCollectUpdateCount(String schema, String sql, Object[] arg) {
+    private CompletableFuture<Long> executeAndCollectUpdateCount(long connectionId, String sql, Object[] arg) {
         var context = createQueryContext(JdbcStatementType.UPDATE_STATEMENT_TYPE);
 
-        var cursors = processor.queryAsync(context, schema, sql, arg);
-
-        if (cursors.size() != 1) {
-            return CompletableFuture.failedFuture(new IgniteInternalException("Multi statement queries are not supported in batching"));
+        JdbcConnectionContext connectionContext;
+        try {
+            connectionContext = resources.get(connectionId).get(JdbcConnectionContext.class);
```

Comment:
May be it would be better to move this code into a separate method and use it from `queryAsync` too.